### PR TITLE
55 use a sql database for storing user preferences 2

### PR DESCRIPTION
**Pull request by louisevelayo**
### Description
This PR is to add the DB API to be able to make SQL queries to the postgreSQL DB instance on Digital Ocean.

### Changes Made
- Implementation of the API is in `node_provider_db.py`
- `node_monitor.py`, `__main.py__` and `test_node_monitor.py` were modified to use the real databse instead of the `tmp_prod_db.json` file.
- `test.yml` was modified to have the database environment variables
- `docker-compose.yml` was modified to remove the binding of the `tmp_prod_db` file.

### Testing
`TEST=true docker compose up --build`
- 8 tests passed, 1 test skipped. This is the same result as when the mock database was being used

### Additional Comments
n/a

### CODE REVIEW

**Comment by mourginakis** [comment]
I changed the DB schema to something I think is a lot more readable, and I implemented what I think is the complete interface for the `NodeProviderDB` class. Type annotations will have to be updated in the future.

Also created a test file.

Deleted a decent amount of code, sorry. Obviously it will remain in previous git commits, so you should still be able to reference it quite easily should you need it. We need all these new methods to be defined. Anything that's currently incomplete raises `NotImplementedError`.

I also created a test file, but just like the class itself, it is only partially complete. We will need to finish both of these over the next couple days. These tests make live CRUD operations to the database, which is cool, but will always fail in Github because we haven't created proper credentials in the `.env` file to run in our workflow.

Perhaps we create a `--db` option in `conftest.py` to do live testing with the database. Otherwise, skip.

**Comment by mourginakis** [comment]
I'm going to revert these two commits.
- `aeca8da`: UNIQUE fields are not needed and could be a source for errors later on if there is clashing data.
- `8f9ecfc`: This code is hard to read

**Comment by mourginakis** [comment]
In [c3bc453](https://github.com/aviate-labs/node-monitor/pull/67/commits/c3bc4536a3c1261c2def82aa9aa4b4e548e36af1) I removed conflict handling that you just implemented. 

Tables `email_lookup` and `channel_lookup` use surrogate keys and duplicate information is fine.

**Comment by louisevelayo** [comment]
> 



> I'm going to revert these two commits for the following reasons.
> 
> * `aeca8da`: UNIQUE fields are not needed and could be a source for errors later on if there is clashing data.
> * `8f9ecfc`: This code is hard to read



> In [c3bc453](https://github.com/aviate-labs/node-monitor/pull/67/commits/c3bc4536a3c1261c2def82aa9aa4b4e548e36af1) I removed the conflict handling that you just implemented. Tables `email_lookup` and `channel_lookup` use surrogate keys and duplicate information is fine.

There is a potential problem with this when it comes to deleting data. 

For example, `delete_channel_lookup('dummy-principal-1')` will delete all rows that have 'dummy-principal-1' in the `node_provider_id` field. So, more rows will be deleted than intended, or you will unknowingly delete some other records. This will be the same for the `email_lookup` table. Is this side-effect ok?

If not, there are two approaches I can think of that I would like to get your thoughts on @mourginakis :
- Add unique constraints to the `email_address` field and the `node_provider_id` field in `email_lookup` and `channel_lookup` respectively (but from previous discussions, this was not preferred)
- Modify the delete functions to take in more parameters to ensure that only one specific row is deleted.

**Comment by mourginakis** [comment]
I was intending to do it the 2nd way you suggested, but we can fix this after we get proper tests written.

**Comment by louisevelayo** [comment]
Commit `fc7a99d` passes all tests locally but not the checks on github actions. This is because it fails to connect to a pgsql database. I'm thinking to either remove the assertions or set up a test pgsql database instance somewhere so that the checks can run properly.

File: node_monitor/node_provider_db.py [hunk 1]
```diff
@@ -315,17 +315,44 @@ def get_channels(self) -> List[Tuple[Any, ...]]:
...
+        """
+        values = (node_label, node_id)
+        self.connect()
+        assert self.conn is not None
+        with self.conn.cursor() as cur:
+            cur.execute(query, values)
+        self.disconnect()
     
-    def _delete_node_label(self) -> None:
+    def _delete_node_label(self, node_label: str) -> None:
```
Comment (mourginakis) [inline node_monitor/node_provider_db.py:333]: This should work on `node_id` and not `node_label`

**Comment by mourginakis** [comment]
Tests for the database are off by default now, so every push should pass CI.
To run these tests:
```python
pytest -s --db tests/test_node_provider_db.py
```

File: node_monitor/node_provider_db.py (hunk 1 above)
Comment (louisevelayo) [reply node_monitor/node_provider_db.py]: I saw that you already changed this so I will resolve this comment :) Thanks for spotting and fixing it!!

**Pull request merged**